src/lib/metrics.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const config = require('../config');
+const logger = require('./bunyan').logger;
+
+const metricsConfig = config?.metrics || {};
+const metricsEnabled = metricsConfig?.enabled || false;
+
+const metricsPrefix = `${
+	metricsConfig?.prefix ? `${metricsConfig.prefix}_` : ''
+}`;
+
+/** @type {import('prom-client')} */
+let Prometheus;
+
+/** @type {import('prom-client').Registry} */
+let register;
+
+/** @type {import('prom-client').Counter} */
+let requestCounter;
+
+/** @type {import('prom-client').Histogram} */
+let requestDurationHistogram;
+
+/**
+ * Initialize the Prometheus registry
+ */
+const start = () => {
+	// Bypass start function if metrics are disabled
+	if (!metricsEnabled) {
+		return;
+	}
+
+	logger.info('Initializing Prometheus Exporter');
+
+	Prometheus = require('prom-client');
+	register = new Prometheus.Registry();
+
+	// If configured, will collect and return common Node specific metrics
+	if (metricsConfig?.collectDefaultMetrics || false) {
+		const collectDefaultMetrics = Prometheus.collectDefaultMetrics;
+		collectDefaultMetrics({ register });
+	}
+
+	_initializeRequestMetrics();
+};
+
+/**
+ * Retrieve current metrics
+ * @returns {Promise<string>}
+ */
+const getMetrics = () => {
+	if (!metricsEnabled) {
+		return Promise.reject('Metrics not configured');
+	}
+
+	return register.metrics();
```

Comment:
@gu - how should we handle situations where the `start()` method is not called prior to a `getMetrics()` request? Understanding that this is now in the middleware initialization, we may want to provide a more explicit error message.